src/clients/FtpRelay.cc

```diff
@@ -385,9 +385,16 @@ Ftp::Relay::forwardReply()
     reply->sources |= Http::Message::srcFtp;
 
     setVirginReply(reply);
-    markParsedVirginReplyAsWhole("Ftp::Relay::handleControlReply() does not forward partial replies");
     adaptOrFinalizeReply();
 
+#if USE_ADAPTATION
+    if (adaptationAccessCheckPending) {
```

Comment:
FwdState::markStoredReplyAsWhole() will be called from Client::handleAdaptedHeader(), i.e., if adaptation is successful. If it is unsuccessful (Adaptation::Answer::akError), storedWholeReply_ remains unset, and there is no assertion in FwdState::completed().